### PR TITLE
Created user solutions pages

### PR DESCRIPTION
**Pull request by Vinceyou1**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
wait so can you show on the site what this does?

**Comment by Vinceyou1** [comment]
This was for #3778 and #2971. It modifies the problem-solutions page to a /problems/user-solutions/[uniqueId] so the user solutions page can be opened in a new tab w/out using the state passed in from the problem list section

**Review by thecodingwizard — CHANGES_REQUESTED** [review]

File: src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx [hunk 1]
```diff
@@ -76,11 +76,7 @@ export default function ProblemsListItemDropdown(
   const [isDropdownShown, setIsDropdownShown] = React.useState(false);
   const onViewProblemSolutions = () => {
     tippyRef.current.hide();
-    navigate('/problem-solutions/', {
-      state: {
-        problem,
-      },
-    });
+    navigate('/problems/user-solutions/' + problem.uniqueId);
```
[the remark this answers is not in the document]
Comment (Vinceyou1) [reply src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx:79]: Internal Solution uses an anchor tag and opens in a new tab by default, do you want me to make the user solutions link consistent with that or just use a Link element?

[the remark this answers is not in the document]
Comment (thecodingwizard) [reply src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx]: Oh, good catch. I think having an anchor tag that opens a new tab by default is better in order to be consistent with user solutions link

File: src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx [hunk 2]
```diff
@@ -45,13 +44,13 @@ function ViewSolutionsContent({
...
   }
   return (
-    <button
-      type="button"
+    <Link
       className="focus:outline-none block w-full text-left px-4 py-2 text-sm text-gray-700 dark:text-gray-300 hover:bg-gray-100 dark:hover:bg-gray-800 hover:text-gray-900"
-      onClick={onViewProblemSolutions}
+      to={`/problems/${problem.uniqueId}/user-solutions`}
+      target="_blank"
     >
```
Comment (thecodingwizard) [inline src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx:51]: can you still use `<Link>` with target blank or do you just use `<a>` here?

File: src/templates/userSolutionTemplate.tsx [hunk 3]
```diff
@@ -4,14 +4,12 @@ import ProblemSolutions from '../components/ProblemSolutions';
...
-) {
+export default function Template(props) {
   const [isSubmitModalOpen, setIsSubmitModalOpen] = React.useState(false);
 
-  // Note: In the future, maybe this page should be generated by gatsby
-  // or we can import all the problems as a big JSON file or something.
-  const problem: ProblemInfo = props.location.state?.problem;
+  // problem isn't passed as type of ProblemInfo from gatsby-node,
+  // but all this page's dependencies needs is the problem's name and uniqueId
+  const problem: ProblemInfo = props.pageContext.problem;
```
Comment (thecodingwizard) [inline src/templates/userSolutionTemplate.tsx:12]: if this is the case, can we change the type from `ProblemInfo` to

```
{ name: string, uniqueId: string }
```

or something?

File: src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx (hunk 2 above)
Comment (Vinceyou1) [reply src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx:51]: Both function the same, lmk if you want to change it to an anchor tag instead

Comment (thecodingwizard) [reply src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx:51]: Maybe change it just to be safe? https://github.com/gatsbyjs/gatsby/issues/13825 (also on the preview deployment it doesn't seem to open in a new tab)

**Review by thecodingwizard — CHANGES_REQUESTED** [review]
Can we also get rid of the "Back" button on the user solutions page? ex. https://usaco-guide-3iqqe7taz-cpinitiative.vercel.app/problems/ys-ZAlgorithm/user-solutions

Thanks!

**Review by thecodingwizard — APPROVED** [review]

**Comment by thecodingwizard** [comment]
Thanks! I'll wait for CI to finish and then merge this.

**Pull request merged**